qpc/report/merge.py

```diff
@@ -120,34 +97,69 @@ def _get_report_ids(self):
+        """
+        # pylint: disable=too-many-branches,too-many-statements
         print(_(messages.REPORT_VALIDATE_JSON % files))
         all_sources = []
+        report_version = None
+        report_type = None
         report_id = None
         for file in files:
-            error = True
             if os.path.isfile(file):
+                details_report = None
                 with open(file) as lint_f:
                     try:
-                        json_data = json.load(lint_f)
+                        details_report = json.load(lint_f)
                     except json_exception_class:
                         print(_(messages.REPORT_JSON_DIR_FILE_FAILED % file))
                         continue
-                    sources = json_data.get('sources')
+
+                    # validate report type
+                    report_type = details_report.get('report_type', None)
+                    if not report_type:
+                        print(_(messages.REPORT_MISSING_REPORT_VERSION % file))
+                    if report_type and report_type != 'details':
+                        print(_(messages.REPORT_INVALID_REPORT_TYPE %
+                                report_type))
+                        continue
+
+                    # validate sources
+                    sources = details_report.get('sources', None)
                     if sources:
-                        facts = sources[0].get('facts')
-                        server_id = sources[0].get('server_id')
-                        if facts and server_id:
-                            error = False
-                if not error:
-                    print(_(messages.REPORT_JSON_DIR_FILE_SUCCESS % file))
-                    try:
-                        sources = self.sources(file)
-                        all_sources += sources
-                    except ValueError:
-                        print(_(messages.REPORT_INVALID_JSON_FILE % file))
-                        sys.exit(1)
-                else:
-                    print(_(messages.REPORT_JSON_DIR_FILE_FAILED % file))
+                        has_error = False
+                        for source in sources:
+                            facts = source.get('facts')
+                            server_id = source.get('server_id')
+                            if not facts:
+                                print(_(messages.REPORT_JSON_MISSING_FACTS %
+                                        file))
+                                has_error = True
+                                break
+                            if not server_id:
+                                print(
+                                    _(messages.REPORT_JSON_MISSING_SERVER_ID %
+                                      file))
+                                has_error = True
+                                break
+                        if not has_error:
+                            if not report_version and \
```

Comment:
I don't understand this check. Could you explain it to me? From what I can tell, report_version can only be None at this point, and if all we are doing if this passes, is setting the report_version, why do we need this check? Ie. why can't we just do:
```report_version = details_report.get('report_version', None)```

Reply:
Take a look again with new locals